packages/eslint-plugin/src/rules/shorthand-property-sorting/index.ts

```diff
@@ -0,0 +1,124 @@
+    return parent;
+  }
+  return findProgramRoot(parent);
+};
+
+export const shorthandFirst: Rule.RuleModule = {
+  meta: {
+    docs: {
+      recommended: true,
+      description:
+        'When using both shorthand and longhand properties, compiled scrambles the properties and it is unintuative to understand which properties override which others. Ordering the shorthand first ensures that longhand overrides shorthand.',
+      url: 'https://github.com/atlassian-labs/compiled/tree/master/packages/eslint-plugin/src/rules/shorthand-property-sorting',
+    },
+    messages: {
+      'shorthand-first':
+        'When using both shorthand and longhand properties, the shorthand property should be first.',
+    },
+    type: 'problem',
+    fixable: 'code',
+  },
+  create(context) {
+    const selectorString: string =
+      'CallExpression[callee.type="Identifier"][callee.name="css"] > ObjectExpression,' +
+      'CallExpression[callee.type="Identifier"][callee.name="cssMap"] > ObjectExpression,' +
+      'CallExpression[callee.type="Identifier"][callee.name="xcss"] > ObjectExpression,' +
+      'CallExpression[callee.type="MemberExpression"][callee.object.name="styled"] > ObjectExpression';
+
+    return {
+      [selectorString]: (node: Rule.Node) => {
+        if (node.type === 'ObjectExpression') {
+          if (node.properties.length > 0) {
+            let lowestDepth = 0;
+            let fixRequired = false;
+            let importedFromCorrectPackage = null;
+            const program = findProgramRoot(node);
+
+            // make sure node is imported from the correct package
+            const logicCssAndCssMap =
+              node.parent.type === 'CallExpression' &&
+              node.parent.callee.type === 'Identifier' &&
+              (node.parent.callee.name === 'css' || node.parent.callee.name === 'cssMap');
+
+            const logicStyled =
+              node.parent.type === 'CallExpression' &&
+              node.parent.callee.type === 'MemberExpression' &&
+              node.parent.callee.object.type === 'Identifier' &&
+              node.parent.callee.object.name === 'styled';
+
+            const logicXcc =
+              node.parent.type === 'CallExpression' &&
+              node.parent.callee.type === 'Identifier' &&
+              node.parent.callee.name === 'xcss';
+
+            if (program && (logicCssAndCssMap || logicStyled)) {
+              importedFromCorrectPackage = program.body.find(
+                (n) => n.type === 'ImportDeclaration' && n.source.value === '@compiled/react'
+              );
+            }
+            if (program && logicXcc) {
+              importedFromCorrectPackage = program.body.find(
+                (n) => n.type === 'ImportDeclaration' && n.source.value === '@atlaskit/primitives'
+              );
+            }
+
+            // loop through the css properties of a ObjectExpression object
+            node.properties.some((property) => {
```

Comment:
add something like `if (importedFromCorrectPackage) { return; }` above the `node.properties.some((property) => { ... }` loop, so we don't needlessly loop over every non-Compiled style usage